amplify/data/resource.ts

```diff
@@ -80,18 +76,56 @@ const schema = a
         allow.group("Admin"),
         allow.authenticated().to(["read"]),
       ]),
-
     Team: a
       .model({
         name: a.string(),
         id: a.id(),
         approved: a.boolean(),
         members: a.hasMany("User", "teamId"),
+        scores: a.hasMany("Score", "teamId"),
+        teamRooms: a.hasMany("TeamRoom", "teamId"),
+      })
+      .authorization((allow) => [
+        allow.group("Admin").to(["read", "update"]),
+        allow.authenticated().to(["read"]),
+      ]),
+    Score: a
+      .model({
+        id: a.id().required(),
+        score: a.integer().required(),
+        judgeId: a.id().required(),
+        judge: a.belongsTo("User", "judgeId"),
+        teamId: a.id().required(),
+        team: a.belongsTo("Team", "teamId"),
+      })
+      .authorization((allow) => [
+        allow.group("Admin").to(["create", "read", "update", "delete"]),
+        allow.group("Judge").to(["create", "update"]),
+      ]),
+    Room: a
+      .model({
+        id: a.id().required(),
+        name: a.string().required(),
+        teamRoom: a.hasMany("TeamRoom", "roomId"),
+        judges: a.hasMany("User", "JUDGE_roomId"),
       })
       .authorization((allow) => [
-        allow.owner().to(["read", "update"]),
+        allow.group("Admin").to(["read", "update"]),
         allow.authenticated().to(["read"]),
       ]),
+    // For handling a many to many relationship of teams and rooms
+    TeamRoom: a
```

Comment:
Is there no field for "zoom link" or anything because we are integrating a video call in our app as well?

Reply:
Good catch, will add that